File: BOJ/1021_회전하는 큐.py
from collections import deque
import sys
input = sys.stdin.readline
N,M = map(int,input().split())
pops=list(map(int,input().split()))
nums,cnt = deque([n+1 for n in range(N)]),0
for p in pops:
    while True:
        if nums[0] == p :
            nums.popleft()
            break
        nums.rotate(-1 if nums.index(p)<=len(nums)//2 else 1)
        cnt +=1
print(cnt)

File: BOJ/1021_회전하는큐.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "../input.txt")
  .toString()
  .split("\n")
let [N, M] = input.shift().split(" ").map(Number)
let nums = input.shift().split(" ").map(Number)
let q = Array.from({ length: N }, (_, i) => i + 1)
let cnt = 0
for (const num of nums) {
  // console.log(num)
  while (1) {
    if (q[0] === num) {
      q.shift()
      break
    } else if (q.indexOf(num) <= q.length / 2) q.push(q.shift())
    else q.unshift(q.pop())
    cnt++
    // console.log(q)
  }
}
console.log(cnt)
